src/lib.rs

```diff
@@ -245,6 +245,7 @@
 #![cfg_attr(rustbuild, feature(staged_api, rustc_private))]
 #![cfg_attr(rustbuild, unstable(feature = "rustc_private", issue = "27812"))]
 #![deny(missing_debug_implementations, missing_docs, warnings)]
+#![allow(clippy::needless_doctest_main)]
```

Comment:
The main functions are not needed in doc tests but they make sense here as they reflect real-world usage more accurately.